server/entity/splashable.go

```diff
@@ -10,6 +10,20 @@ import (
 	"time"
 )
 
+// SplashableBlock is a block that can be splashed with a splash bottle.
+type SplashableBlock interface {
+	world.Block
+	// Splash is called when a type that implements splashable splashes onto a block.
+	Splash(w *world.World, pos cube.Pos, p potion.Potion)
+}
+
+// SplashableEntity is an entity that can be splashed with a splash bottle.
+type SplashableEntity interface {
+	world.Entity
+	// Splash is called when a type that implements splashable splashes onto an entity.
+	Splash(w *world.World, pos mgl64.Vec3, p potion.Potion)
```

Comment:
The `potion.Potion`  passed will always be `potion.Water()`. I'm not entirely sure if there is much of a need for this outside of water splash potions though, so maybe just get rid of the third arg?

cc @TwistedAsylumMC

Reply:
I keep this the same until seb reviews it but I don't see a use for it either.

Reply:
Happy for the 3rd arg to be removed here

Reply:
Do the same for the SplashableBlock interface